src/components/project/project-step.resolver.ts

```diff
@@ -3,24 +3,37 @@ import { stripIndent } from 'common-tags';
 import { AnonSession, ID, Session } from '../../common';
 import { ProjectStepTransition, SecuredProjectStep } from './dto';
 import { ProjectRules } from './project.rules';
+import { ProjectService } from './project.service';
 
 @Resolver(SecuredProjectStep)
 export class ProjectStepResolver {
-  constructor(private readonly projectRules: ProjectRules) {}
+  constructor(
+    private readonly projectRules: ProjectRules,
+    private readonly projectService: ProjectService,
+  ) {}
 
   @ResolveField(() => [ProjectStepTransition], {
     description: 'The available steps a project can be transitioned to.',
   })
   async transitions(
-    @Parent() step: SecuredProjectStep & { parentId: ID; changeset?: ID },
+    @Parent()
+    step: SecuredProjectStep & {
+      parentId: ID;
+      changeset?: ID;
+    },
     @AnonSession() session: Session,
   ): Promise<ProjectStepTransition[]> {
     if (!step.canRead || !step.canEdit || !step.value) {
       return [];
     }
+    const project = await this.projectService.readOneUnsecured(
```

Comment:
@CarsonF - I know that you mentioned using a loader here.  I'm not entirely sure what would be used for the key for the loader or how that would work.  We looked at some other examples, but couldn't quite get there.

Reply:
https://github.com/SeedCompany/cord-api-v3/blob/dfb31cc2a5922d776695c11582e5d447d772ad22/src/components/product/product.resolver.ts#L84-L84
```ts
const project = await projects.load(step.parentId)
```

Reply:
I tried this, but step.parentId is of type ID and the load method says it receives a type of 'key'.  I wasn't sure how to transform the ID argument into a type of 'key'

Reply:
Ah right sorry
```ts
load({ id: step.parentId, view: { active: true } })
```

Reply:
ok, thank you sir

Reply:
It's the `view` part we didn't really get I think.

Reply:
It's apart of the changesets work. "active" is the normal view, not inside a changeset. It's only half baked, and being dropped in edgedb migration.